pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -363,7 +363,13 @@ private BrokerResponseNative handleSQLRequest(long requestId, String query, Json
 
     // Validate the request
     try {
-      validateRequest(pinotQuery, _queryResponseLimit);
+      int numReplicas = 1;
+      if (offlineTableConfig != null) {
+        numReplicas = offlineTableConfig.getValidationConfig().getReplicationNumber();
+      } else if (realtimeTableConfig != null) {
+        numReplicas = realtimeTableConfig.getValidationConfig().getReplicationNumber();
```

Comment:
For realtime, `getReplicasPerPartitionNumber()` is the correct indicator for the replication factor.

Reply:
removed this , no longer using numReplicas to check and pass to validateRequest